ctapipe_io_magic/__init__.py

```diff
@@ -1410,11 +1432,19 @@ def _load_data(self):
             # (set in the DAQ code), we create an artificial event ID by using the subrun number and
             # attaching the DAQ event id padded with 0 (to have 5 digits for the DAQ event id).
             # Like this we obtain an event id which is unique within a data run (like fStereoEvtNumber).
-            if self.is_stereo:
-                calib_data[event_type]['event_number'] = np.array(common_info['MRawEvtHeader.fStereoEvtNumber'], dtype=int)
+
+            if self.is_mc:
```

Comment:
maybe this would work instead of all those conditions:
```
if (self.is_mc and self.use_mc_mono_events) or not self.is_stereo:
  subrun_id = self.uproot_file["RunHeaders"]['MRawRunHeader.fSubRunIndex'].array(library="np")[0]
  calib_data[event_type]['event_number'] = np.array([f"{subrun_id}{daq_id:05}" for daq_id in common_info['MRawEvtHeader.fDAQEvtNumber']], dtype=int)
else:
  calib_data[event_type]['event_number'] = np.array(common_info['MRawEvtHeader.fStereoEvtNumber'], dtype=int)
```